internal/api/v1/queryparams/query_params_builder.go

```diff
@@ -0,0 +1,109 @@
+package queryparams
+
+import (
+	"errors"
+	"net/url"
+
+	goclienterr "github.com/bitcoin-sv/spv-wallet-go-client/errors"
+	"github.com/bitcoin-sv/spv-wallet-go-client/queries"
+	"github.com/bitcoin-sv/spv-wallet/models/filter"
+)
+
+type Parser interface {
+	Parse() (url.Values, error)
+}
+
+type Builder struct {
+	parsers []Parser
+}
+
+func (b *Builder) Build() (*URLValues, error) {
+	params := NewURLValues()
+	for _, p := range b.parsers {
+		values, err := p.Parse()
+		if err != nil {
+			return nil, errors.Join(err, goclienterr.ErrQueryParamsBuildFailed)
+		}
+
+		if len(values) > 0 {
+			params.Append(values)
+		}
+	}
+
+	return params, nil
+}
+
+func NewBuilder[F queries.QueryFilters](query *queries.Query[F]) (*Builder, error) {
+	var parsers []Parser
+	if query.Metadata != nil {
+		parsers = append(parsers, &MetadataParser{Metadata: query.Metadata, MaxDepth: DefaultMaxDepth})
+	}
+
+	if query.PageFilter != (filter.Page{}) {
+		parsers = append(parsers, &FilterParser{Filter: query.PageFilter})
+	}
+
+	parsers = append(parsers, initNonAdminQueryFilters(query)...)
+	parsers = append(parsers, initAdminQueryFilters(query)...)
```

Comment:
I think here we also are adding two filters twice? For example let's take AccessKeysFilter into the consideration the first filter will add: 
`[]Parser{&FilterParser{Filter: f.ModelFilter}, &FilterParser{Filter: f}}`
while the second admin query filter will also append:
`[]Parser{
			&FilterParser{Filter: f.AccessKeyFilter},
			&FilterParser{Filter: f.ModelFilter},
			&FilterParser{Filter: f}}`

And we have duplicate filters for `Filter: f.ModelFilter && Filter: f`
Not sure if it's by design LMK